The favorite icon and favorite actions were created.

mselmanaslan opened the pull request:
A service class was created using SQLite for favorite actions.

The Favorite Icon was prepared as a component and used within the cell.

A callback was added to the Table element for the Cells to be used later.

Commit:
The favorite icon and favorite actions were created.

A service class was created using SQLite for favorite actions.

The Favorite Icon was prepared as a component and used within the cell.

A callback was added to the Table element for the Cells to be used later.

yusufkaran commented on RMCharactersAppUIKit/Network/Service/FavoriteService.swift:
```diff
@@ -0,0 +1,172 @@
+                )
+                """)
+        } catch {
+            fatalError("Error creating table: \(error)")
```
Kullanıcı açısından fatalError ile programı sonlandırmak yerine throw kullanarak hata fırlatabilirsin

Commit:
Refactor database initialization and table creation to use throwing structure

The createTable() method was updated to use a throwing structure instead of fatalError.
